libfreerdp/core/gateway/rdg.c

```diff
@@ -253,15 +256,15 @@ static BOOL rdg_read_http_unicode_string(wStream* s, WCHAR** string, UINT16* len
 	Stream_Read_UINT16(s, strLenBytes);
 
 	/* Remember position of our string */
-	Stream_GetPointer(s, str);
+	Stream_GetPointer(s, str.b);
```

Comment:
This union magic looks like a hack and makes the code hard to read - why don't we use a simple cast like `str = (WCHAR*)Stream_Pointer(s);`